### PR TITLE
Self-Service Fork Request: PR Workflow

### PR DESCRIPTION
**Pull request by lelia**
## Description

With https://github.com/wayfair/ospo-automation/pull/14 paving the path for semi-automated PR templates, this PR aims to fully productionize the POC introduced in https://github.com/wayfair/ospo-automation/pull/8 in order to make the `Fork a repository` [PR template](https://github.com/wayfair/ospo-automation/blob/self-service-fork-workflow/.github/PULL_REQUEST_TEMPLATE/fork-repo.md) fully self-service for Wayfair developers. This PR also depends on the `forkUrl` output introduced in [forker v0.0.6](https://github.com/wayfair-incubator/forker/releases/tag/v0.0.6) (via https://github.com/wayfair-incubator/forker/pull/230). 

In order to test the functionality of this automated workflow, I created several 'dummy' request YAML files to ensure that the validation checks and other parsing logic worked correctly, hence the many PR approvals and fork creation comments you'll notice were left by `lhasa-ospo`.

For reviewing purposes, it may be helpful to follow along in the markdown-rendered [README for forking a repository](https://github.com/wayfair/ospo-automation/blob/self-service-fork-workflow/requests/fork/README.md) (this is also linked to in the main README), to make sure the instructions are clear and easy to follow. If you'd like to test out the workflow yourself, I _believe_ you may be able to do so by filing a PR against this branch (`self-service-fork-worfklow`) and following the aforementioned README instructions.

## Type of Change

- [ ] Bug Fix
- [x] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/wayfair/ospo-automation/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [x] Functionality is documented
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [x] All new and existing tests pass

🚨 Not the PR template you wanted? Click the `Preview` tab above, then navigate back to the [OSPO Automation PR Menu](?expand=1)!

### CODE REVIEW

**Review by lhasa-ospo — APPROVED** [review]

**Review by lhasa-ospo — APPROVED** [review]

**Comment by lhasa-ospo** [comment]
🎉 All set @lelia!   Here's your fork: https://github.com/wayfair-contribs/typescript-action

**Review by lhasa-ospo — APPROVED** [review]

**Comment by lhasa-ospo** [comment]
🎉 All set @lelia!   Here's your fork: https://github.com/wayfair-contribs/ort

**Comment by lhasa-ospo** [comment]
🎉 All set @lelia!   Here's your fork: https://github.com/wayfair-contribs/auto-approve-action

**Comment by lhasa-ospo** [comment]
🚨 Hey @lelia, looks like there was an issue processing your fork request!   Please ensure you've committed a properly-formatted YAML (.yml) file to the 'requests/fork/' directory.

**Comment by lhasa-ospo** [comment]
🚨 Hey @lelia, looks like there was an issue processing your fork request!   Please ensure you've committed a properly-formatted YAML (`.yml``) file to the `requests/fork/` directory.

**Comment by lhasa-ospo** [comment]
🚨 Hey @lelia, looks like there was an issue processing your fork request!   Please ensure you've committed a properly-formatted YAML (`.yml`) file to the `requests/fork/` directory.

**Review by lhasa-ospo — APPROVED** [review]

**Comment by lhasa-ospo** [comment]
🎉 All set @lelia!   Here's your fork: https://github.com/wayfair-contribs/agent

File: .github/PULL_REQUEST_TEMPLATE/fork-repo.md [hunk 1]
```diff
@@ -4,7 +4,9 @@
 
 ## Description
 
-This PR template is intended for requesting [repository forks](https://docs.github.com/en/get-started/quickstart/fork-a-repo) into the [`/wayfair-contribs`](https://github.com/wayfair-contribs) organization, to enable upstream contributions to existing open source projects. **You must already be a GitHub organization member of `/wayfair-contribs`** (and preferrably, [`/wayfair`](https://github.com/wayfair) + [`/wayfair-incubator`](https://github.com/wayfair-incubator) as well) **in order to use this workflow**. The repository you are requesting a fork of must also comply with our current list of approved open source licenses: **MIT**, **BSD** (3-clause, 2-clause, 0-clause), **Apache** (1.0, 1.1, 2.0).
+This PR template is intended for Wafair developers requesting [repository forks](https://docs.github.com/en/get-started/quickstart/fork-a-repo) into the [`/wayfair-contribs`](https://github.com/wayfair-contribs) organization, to enable upstream contributions to existing open source projects.
```
Comment (draco2003) [inline .github/PULL_REQUEST_TEMPLATE/fork-repo.md:7]:
```suggestion
This PR template is intended for Wayfair developers requesting [repository forks](https://docs.github.com/en/get-started/quickstart/fork-a-repo) into the [`/wayfair-contribs`](https://github.com/wayfair-contribs) organization, to enable upstream contributions to existing open source projects.
```

**Comment by lelia** [comment]
Note that the fork workflow is going to fail on any new commits since my branch no longer has a `.yml` file under `requests/fork`, this shouldn't prevent the PR from being merge-able, though.

**Comment by lhasa-ospo** [comment]
🚨 Hey @lelia, looks like there was an issue processing your fork request!   Please ensure you've committed a properly-formatted YAML (`.yml`) file to the `requests/fork/` directory.

**Comment by lelia** [comment]
> 🚨 Hey @lelia, looks like there was an issue processing your fork request! Please ensure you've committed a properly-formatted YAML (`.yml`) file to the `requests/fork/` directory.

Note that the [`fork`](https://github.com/wayfair/ospo-automation/actions/workflows/fork.yml) workflow is going to fail on any new commits since my branch no longer has a `.yml` file under `requests/fork`, this shouldn't prevent the PR from being merge-able, though.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,13 +1,26 @@
...
-🚨 INSTRUCTIONS: Click the `Preview` tab above, then select your desired PR template below!
+🚨 INSTRUCTIONS: Click the `Preview` tab above, then follow along below!
 
-## Self-Service Requests
+## Description
 
-- ✅ [Fork a repository](?expand=1&template=fork-repo.md)
-- ❌ [Create a new repository](?expand=1&template=create-repo.md)
-- ❌ [Migrate an existing repository](?expand=1&template=migrate-repo.md)
+This PR template is intended for Wafair developers requesting [repository forks](https://docs.github.com/en/get-started/quickstart/fork-a-repo) into the [`/wayfair-contribs`](https://github.com/wayfair-contribs) organization, to enable upstream contributions to existing open source projects.
```
Comment (GaryPWhite) [inline .github/PULL_REQUEST_TEMPLATE.md:7]:
```suggestion
This PR template is intended for Wayfair developers requesting [repository forks](https://docs.github.com/en/get-started/quickstart/fork-a-repo) into the [`/wayfair-contribs`](https://github.com/wayfair-contribs) organization, to enable upstream contributions to existing open source projects.
```

File: .github/workflows/fork.yml [hunk 3]
```diff
@@ -1,46 +1,89 @@
...
+        with:
+          github-token: ${{ secrets.OSPO_API_TOKEN }}
+          script: |
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: "🚨 Hey @${{ env.GITHUB_ACTOR }}, looks like there was an issue processing your fork request! \
+              Please ensure you've committed a properly-formatted YAML (`.yml`) file to the `requests/fork/` directory."
+            })
```
Comment (GaryPWhite) [inline .github/workflows/fork.yml:45]: NEAT

**Review by GaryPWhite — APPROVED** [review]

File: .github/workflows/fork.yml (hunk 3 above)
Comment (lelia) [reply .github/workflows/fork.yml:45]: YAS

**Comment by lhasa-ospo** [comment]
🚨 Hey @lelia, looks like there was an issue processing your fork request!   Please ensure you've committed a properly-formatted YAML (`.yml`) file to the `requests/fork/` directory.

**Review by lhasa-ospo — APPROVED** [review]

**Comment by lhasa-ospo** [comment]
🎉 All set @lelia!   Here's your fork:

**Review by lhasa-ospo — APPROVED** [review]

**Comment by lhasa-ospo** [comment]
🎉 All set @lelia!   Here's your fork: https://github.com/wayfair-contribs/filters

**Pull request merged**